File: backend/routes/projects.js
import express from 'express';
import Project from '../models/Project.js';
import connectMongo from '../lib/mongodb.js';

const router = express.Router();

router.get('/', async (req, res) => {
  await connectMongo();
  const project = await Project.findOne({ slug: 'default' });
  return res.status(200).json({ diagrams: project?.diagrams || [] });
});

router.post('/', async (req, res) => {
  await connectMongo();
  const { name, description, cloudinaryUrl, repoUrl } = req.body;
  const project = await Project.findOneAndUpdate(
    { slug: 'default' },
    {
      slug: 'default',
      name: 'Agentverse UML Project',
      repoUrl,
      $push: { diagrams: { name, description, cloudinaryUrl, createdAt: new Date() } }
    },
    { upsert: true, new: true }
  );
  return res.status(201).json({ project });
});

router.all('*', (req, res) => {
  res.status(405).json({ message: 'Method not allowed' });
});

export default router;

File: backend/routes/projects.py
from datetime import datetime

from datetime import datetime

from fastapi import APIRouter, HTTPException
from pymongo import ReturnDocument
from backend.models import ProjectPayload
from backend.database import db

router = APIRouter()
collection = db['projects']

@router.get('/')
def get_projects():
    project = collection.find_one({'slug': 'default'})
    if not project:
        return {'diagrams': []}

    return {'diagrams': project.get('diagrams', [])}

@router.post('/')
def add_project(payload: ProjectPayload):
    if not payload.name:
        raise HTTPException(status_code=400, detail='Diagram name is required')

    project = collection.find_one_and_update(
        {'slug': 'default'},
        {
            '$set': {
                'slug': 'default',
                'name': 'Agentverse UML Project',
                'repoUrl': payload.repoUrl
            },
            '$push': {
                'diagrams': {
                    'name': payload.name,
                    'description': payload.description,
                    'cloudinaryUrl': payload.cloudinaryUrl,
                    'createdAt': datetime.utcnow()
                }
            }
        },
        upsert=True,
        return_document=ReturnDocument.AFTER
    )

    return {'project': project}
